docs/pipeline_configuration.md

```diff
@@ -0,0 +1,75 @@
+
+### YAML file format
+The file in general has two root keys, the "version", and the "pipelines" map. The map consists of other elements like 
+"status" and "name", which are configurations for the pipeline itself.
+
+To create connectors in that pipeline, simply add another map under the pipeline map, and call it "connectors".
+
+To create processors, either add a "processors" map under a pipeline ID, or under a connector ID, depending on its parent.
+check this YAML file example with explanation for each field:
+
+``` yaml
+version: 1.0                    # parser version, the only supported version for now is 1.0 [mandatory]
+pipelines:                      # a map of pipelines IDs and their configurations.
+  pipeline1:                    # pipeline ID, has to be unique.
+    status: running             # pipelines status at startup, either running or stopped. [mandatory]
+    name: pipeline1             # pipeline name. [optional]
+    description: desc           # pipeline describtion. [optional]
+    connectors:                 # a map of connectors IDs and their configurations.
+      con1:                     # connector ID, has to be unieuq per pipeline.
+        type: source            # connector type, either "source" or "destination". [mandatory]
+        plugin: builtin:file    # connector plugin. [mandatory]
+        name: con3              # connector name. [optional]
+        settings:               # map of configurations keys and their values.
+          path: ./file1.txt     # for this example, the plugin "bultin:file" has only one configuration, which is path.
+      con2:
+        type: destination
+        plugin: builtin:file
+        name: file-dest
+        settings:
+          path: ./file2.txt
+        processors:             # a map of processor IDs and their configurations, "con2" is the processor parent.
+          proc1:                 # processor ID, has to be unique for each parent
+            type: js            # processor type. [mandatory]
+            settings:           # map of processor configurations and values
+              Prop1: string
+    processors:                 # processor IDs, that have the pipeline "pipeline1" as a parent.
+      proc2: 
+        type: js
+        settings:
+          additionalProp1: string
+          additionalProp2: string
+```
+
+### Updates and Deletes
+Updates and deletes for a provisioned pipeline can only be done through the configuration files.
+
+* To delete a pipeline: simply delete it from the pipelines map from the configuration file, then run conduit again.
+* To update a pipeline: change any field value from the configuration file, and run conduit again to address these updates.
+  
+Note: Updates would keep the status of the pipeline and it would continue working from where it stopped. However, 
+  If the Updated configurations were something like "connector-id", "plugin", "pipeline-id", or "processor-id", then 
```

Comment:
We should be very specific about which changes will cause a source connector to lose state. "Something like" sounds very vague and keeps me wondering what else might cause a pipeline to start from the beginning.

Also, I feel like this shouldn't be a "note" but just a normal paragraph.